### PR TITLE
Silence clang7 self-assignment warnings in tests

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: tests/Unit/DataStructures/Test_Variables.cpp [hunk 1]
```diff
@@ -160,7 +160,14 @@ SPECTRE_TEST_CASE("Unit.DataStructures.Variables", "[DataStructures][Unit]") {
   CHECK(get_output(empty_vars) == "Variables is empty!");
 
   // Check self-assignment
+#ifdef __clang__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wself-assign-overloaded"
+#endif //__clang__
   v = v;
+#ifdef __clang__
+#pragma GCC diagnostic push
```
Comment (wthrowe) [inline tests/Unit/DataStructures/Test_Variables.cpp]: pop

File: tests/Unit/TestHelpers.hpp [hunk 2]
```diff
@@ -84,8 +84,15 @@ void test_copy_semantics(const T& a) {
...
   const T c(a);  // NOLINT
   CHECK(c == a);
+#ifdef __clang__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wself-assign-overloaded"
+#endif //__clang__
   // clang-tidy: self-assignment
   b = b;  // NOLINT
+#ifdef __clang__
+#pragma GCC diagnostic push
```
Comment (wthrowe) [inline tests/Unit/TestHelpers.hpp]: pop

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
@wthrowe derp, sorry about that. Fixed now :)

**Review by wthrowe — DISMISSED** [review]

**Review by kidder — DISMISSED** [review]

**Comment by kidder** [comment]
@nilsdeppe this didn't compile in the clang builds

**Comment by nilsdeppe** [comment]
@kidder @wthrowe this should be ready to go now :)

**Review by kidder — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**